core/src/main/scala/tamer/job/AbstractStatefulSourceTamerJob.scala

```diff
@@ -0,0 +1,50 @@
+package tamer.job
+
+import com.sksamuel.avro4s.Codec
+import tamer.config.KafkaConfig
+import tamer.{SourceConfiguration, TamerError}
+import zio.blocking.Blocking
+import zio.clock.Clock
+import zio.{Chunk, Queue, Ref, Schedule, UIO, ZIO}
+
+import java.time.Duration
+
+abstract class AbstractStatefulSourceTamerJob[
```

Comment:
not sure I would call this stateful, the function `next` that is supplied to Kafka.live serves exactly that purpose, maintaining the state (in kafka)

Reply:
this seems like uses some other state, transient...

Reply:
@gurghet suggestions? I don't like stateful (and probably would like to revisit all these def names in it) cause all these connectors are stateful, this one (s3 and likely also object storage) needs an out of band transient state